tests/unit/experimental/pipeline/test_orchestrator.py

```
@@ -164,6 +169,43 @@ async def test_orchestrator_get_component_inputs_ignore_user_input_if_input_def_
         )
 
 
+@pytest.mark.asyncio
+@patch(
+    "neo4j_graphrag.experimental.pipeline.pipeline.Orchestrator.get_status_for_component"
+)
+async def test_orchestrator_set_component_status(mock_status: Mock) -> None:
```

This might be a bit nit-picky but are we covering all possible transitions with this test? Is it possible to go from `RUNNING` back to `UNKNOWN` for example? If not do we want to raise an error as we do with the `DONE` to `RUNNING` transition?

---

Makes sense. Done in https://github.com/neo4j/neo4j-graphrag-python/pull/293/commits/3fdc4004956c26a935c8c5eb8246eb7bdeec7525 (transitionning to a `parametrize` test for readability.